### PR TITLE
fix default spectrum 2d viewer assignment for slit overlay plugin

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request fixes a bug in the slit overlay plugin in Mosviz which resulted in a snackbar error being raised complaining about `S_REGION`.  The underlying cause was that the plugin was accessing data from the image viewer instead of the spectrum-2d-viewer because of faulty default viewer assignment introduced in #1681.  This PR does not change the actual parsing or treatment of the slit itself, which still needs validation/testing.

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/app.py [hunk 1]
```diff
@@ -1167,6 +1167,7 @@ def _get_first_viewer_reference_name(
         )
 
         spectral_viewers = (SpecvizProfileView, CubevizProfileView)
+        spectral_2d_viewers = (MosvizProfile2DView, )
         table_viewers = (MosvizTableViewer, )
         image_viewers = (MosvizProfile2DView, CubevizImageView, SpectralExtraction)
```
Comment (kecnry) [inline jdaviz/app.py:1172]: I would expect this to have been `(ImvizImageView, CubevizImageView)`... are `MosvizProfile2DView` and/or `SpectralExtraction` necessary for any existing plugins to function (@bmorris3)?

Comment (bmorris3) [reply jdaviz/app.py:1172]: I can't be sure without experimenting – have you tried on this branch??

Comment (kecnry) [reply jdaviz/app.py:1172]: `SpectralExtraction` is a plugin, it must have been added for a reason?  I can try and run tests and just make sure plugins don't crash on load, but it's hard to tell if it has other unintended consequences (the slit overlay bug slipped under the radar for quite a while) 😬

Comment (kecnry) [reply jdaviz/app.py]: passes tests and couldn't find any issues when manually testing, but further testing during reviews is welcomed!

**Review by bmorris3 — APPROVED** [review]
Looks good. I'm not sure what confluence of constraints in https://github.com/spacetelescope/jdaviz/pull/1681 led to this in the first place, which is about as good a reminder as any that PRs with `+714 −258` might be too big.

**Comment by javerbukh** [comment]
Seeing the following traceback when selecting a new row in the MosvizExample notebook (NIRSpec):
```
File ~/Documents/jdaviz_dev/jdaviz/jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py:150, in SlitOverlay.remove_slit_overlay(self)
    147 def remove_slit_overlay(self):
    148     if self._slit_overlay_mark is not None:
    149         image_figure = self.app.get_viewer(
--> 150             self._default_flux_viewer_reference_name
    151         ).figure
    152         # We need to do the following instead of just removing directly on
    153         # the marks otherwise traitlets doesn't register a change in the
    154         # marks.
    155         marks = image_figure.marks.copy()

AttributeError: 'SlitOverlay' object has no attribute '_default_flux_viewer_reference_name'
```

**Comment by kecnry** [comment]
@bmorris3 - does that line @javerbukh mentioned just need to be switched to `_default_image_viewer_reference_name`?  Or is there a reason its set to flux viewer?

**Comment by rosteen** [comment]
I just tested with MosvizExample and don't see that traceback, I wonder if something in your environment is out of date @javerbukh. I don't really want to approve until you can confirm it works for you, but it looks good to me.

**Comment by rosteen** [comment]
Any sense of whether @javerbukh's error was a local environment thing and we can merge, or something that needs to be addressed?

**Comment by javerbukh** [comment]
It shows up in the console log, but otherwise the PR works as expected (I can see the slit now). No one else sees it appearing in their console?

**Comment by kecnry** [comment]
I think it likely does need the change I proposed.  @bmorris3 - can you confirm?

**Comment by javerbukh** [comment]
@kecnry Just checked and that fixed the error. @bmorris3, I can approve after that line is changed.

**Comment by bmorris3** [comment]
One (small) inconsistency now. The tray registry expects either `flux` or `image`: 

https://github.com/spacetelescope/jdaviz/blob/9fc2f15580044e3d40695512a7974a61473dc894/jdaviz/core/registries.py#L86-L93

`SlitOverlay` currently tells the tray registry that this object requires a flux viewer:

https://github.com/spacetelescope/jdaviz/blob/5714082de91287e872c4008eee8daedd8115f7c9/jdaviz/configs/mosviz/plugins/slit_overlay/slit_overlay.py#L45-L47

Should that be `image` rather than `flux` for consistency?

**Review by javerbukh — APPROVED** [review]
Works for me now, thank you!

**Pull request merged**